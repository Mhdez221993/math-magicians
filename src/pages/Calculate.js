import React from 'react';
import Calculator from '../components/Calculator';

const Calculate = () => (
  <div className="calculator-wrapper">
    <div className="description">
      <h2>WHAT IS NEW</h2>
      <p>
        Calculator now displays decimal results and symbolic results for
        fractions or certain advanced functions.
        For example, 10 รท 12 will display both 0.83333....
      </p>
    </div>
    <Calculator />
  </div>
);

export default Calculate;
