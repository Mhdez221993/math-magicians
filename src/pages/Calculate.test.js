import React from 'react';
import renderer from 'react-test-renderer';
import { fireEvent, render } from '@testing-library/react';
import Calculate from './Calculate';

describe('test calculator and correct rendering of the calculate page', () => {
  it('Calculate page is rendered correctly', () => {
    const tree = renderer
      .create(<Calculate />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calculator works ( result displayed = 0.5)', () => {
    const { getByText } = render(<Calculate />);

    fireEvent.click(getByText('5'));
    fireEvent.click(getByText('รท'));
    fireEvent.click(getByText('1'));
    fireEvent.click(getByText('0'));
    fireEvent.click(getByText('='));

    expect(getByText('0.5')).toBeInTheDocument();
  });
});
