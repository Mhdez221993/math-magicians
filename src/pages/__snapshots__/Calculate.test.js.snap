// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  className="calculator-wrapper"
>
  <div
    className="description"
  >
    <h2>
      WHAT IS NEW
    </h2>
    <p>
      Calculator now displays decimal results and symbolic results for fractions or certain advanced functions. For example, 10 รท 12 will display both 0.83333....
    </p>
  </div>
  <div
    className="Calculator"
  >
    <div
      className="Display"
    >
      <span
        className="Screen"
      >
        0
      </span>
    </div>
    <div
      className="buttons"
    >
      <div
        className="button"
      >
        <button
          className="btn"
          onClick={[Function]}
          type="button"
          value="AC"
        >
          AC
        </button>
      </div>
      <div
        className="button"
      >
        <button
          className="btn"
          onClick={[Function]}
          type="button"
          value="+/-"
        >
          +/-
        </button>
      </div>
      <div
        className="button"
      >
        <button
          className="btn"
          onClick={[Function]}
          type="button"
          value="%"
        >
          %
        </button>
      </div>
      <div
        className="button"
      >
        <button
          className="btn"
          onClick={[Function]}
          type="button"
          value="รท"
        >
          รท
        </button>
      </div>
      <div
        className="button"
      >
        <button
          className="btn"
          onClick={[Function]}
          type="button"
          value="7"
        >
          7
        </button>
      </div>
      <div
        className="button"
      >
        <button
          className="btn"
          onClick={[Function]}
          type="button"
          value="8"
        >
          8
        </button>
      </div>
      <div
        className="button"
      >
        <button
          className="btn"
          onClick={[Function]}
          type="button"
          value="9"
        >
          9
        </button>
      </div>
      <div
        className="button"
      >
        <button
          className="btn"
          onClick={[Function]}
          type="button"
          value="x"
        >
          x
        </button>
      </div>
      <div
        className="button"
      >
        <button
          className="btn"
          onClick={[Function]}
          type="button"
          value="4"
        >
          4
        </button>
      </div>
      <div
        className="button"
      >
        <button
          className="btn"
          onClick={[Function]}
          type="button"
          value="5"
        >
          5
        </button>
      </div>
      <div
        className="button"
      >
        <button
          className="btn"
          onClick={[Function]}
          type="button"
          value="6"
        >
          6
        </button>
      </div>
      <div
        className="button"
      >
        <button
          className="btn"
          onClick={[Function]}
          type="button"
          value="-"
        >
          -
        </button>
      </div>
      <div
        className="button"
      >
        <button
          className="btn"
          onClick={[Function]}
          type="button"
          value="1"
        >
          1
        </button>
      </div>
      <div
        className="button"
      >
        <button
          className="btn"
          onClick={[Function]}
          type="button"
          value="2"
        >
          2
        </button>
      </div>
      <div
        className="button"
      >
        <button
          className="btn"
          onClick={[Function]}
          type="button"
          value="3"
        >
          3
        </button>
      </div>
      <div
        className="button"
      >
        <button
          className="btn"
          onClick={[Function]}
          type="button"
          value="+"
        >
          +
        </button>
      </div>
      <div
        className="button"
      >
        <button
          className="btn"
          onClick={[Function]}
          type="button"
          value="0"
        >
          0
        </button>
      </div>
      <div
        className="button"
      >
        <button
          className="btn"
          onClick={[Function]}
          type="button"
          value="."
        >
          .
        </button>
      </div>
      <div
        className="button"
      >
        <button
          className="btn"
          onClick={[Function]}
          type="button"
          value="="
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
